lib/assessment.js

```diff
@@ -29,7 +46,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  var myArray = [];
+  for(i=0;i<length;i++){
+    myArray[i] = defaultsFunction(myArray[i]);
```

Comment:
Not quite - the question says that `i` should be the argument, not `myArray[i]`. Additionally, `myArray[i]` doesn't have a value yet, so it doesn't make a ton of sense to pass `undefined` into `defaultsFunction` each time.